src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -19,94 +19,224 @@
+        this.queryText = queryText;
+    }
+
     /**
-     * Construct a new simple query with the given text
+     * For serialization only: Construct a new simple query string query.
+     * 
+     * @param queryText text to parse query from.
+     * @param boost Boost to apply to resulting Lucene query.
+     * @param fields Fields to run query against.
+     * @param analyzer Analyzer to use for this query.
+     * @param queryName Name of the query.
+     * @param operator Default operator to link boolean clauses, OR otherwise.
+     * @param minimumShouldMatch optional minimumShouldMatch to apply to this query should it result in a boolean query.
+     * @param flags Flags to apply, ALL by default.
+     * @param locale Locale to use, ROOT by default.
+     * @param lowercaseExpandedTerms Indicate whether parsed terms should be converted to lowercase, true by default.
+     * @param lenient Should parsing be lenient, false by default.
+     * @param analyzeWildcard false by default.
      */
-    public SimpleQueryStringBuilder(String text) {
-        this.queryText = text;
+    SimpleQueryStringBuilder(String queryText, float boost, Map<String, Float> fields, String analyzer, String queryName, Operator operator,
+            String minimumShouldMatch, int flags, Locale locale, Boolean lowercaseExpandedTerms, Boolean lenient, Boolean analyzeWildcard) {
+        Preconditions.checkNotNull(queryText);
+        this.queryText = queryText;
+        this.boost = boost;
+
+        if (fields != null && fields.size() > 0) {
+            this.fieldsAndWeights.putAll(fields);
+        }
+
+        this.analyzer = analyzer;
+        this.queryName = queryName;
+        this.defaultOperator = operator;
+        this.minimumShouldMatch = minimumShouldMatch;
+        this.flags = flags;
+        this.settings = new Settings(locale, lowercaseExpandedTerms, lenient, analyzeWildcard);
     }
 
     /**
-     * Add a field to run the query against
+     * Construct a new simple query string query.
+     * 
+     * @param queryText Text to parse query from.
+     * @param boost Boost to apply to resulting Lucene query.
+     * @param fields Fields to run query against.
+     * @param analyzer Analyzer to use for this query.
+     * @param queryName Name of the query.
+     * @param operator Default operator to link boolean clauses, OR otherwise.
+     * @param minimumShouldMatch Optional minimumShouldMatch to apply to this query should it result in a boolean query.
+     * @param flags Flags to apply, ALL by default.
+     * @param locale Locale to use, ROOT by default.
+     * @param lowercaseExpandedTerms Indicate whether parsed terms should be converted to lowercase, true by default.
+     * @param lenient Should parsing be lenient, false by default.
+     * @param analyzeWildcard false by default.
      */
+    public SimpleQueryStringBuilder(String text, float boost, Map<String, Float> fields, String analyzer, String queryName, Operator operator,
```

Comment:
Happy to remove. 

We once had the discussion of making all fields in a query builder final, I guess the constructor above is a leftover of that discussion. Seeing the number of parameters for this query alone though made it seem less feasible to move to having all fields final for me.